cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/MetricSampler.java

```diff
@@ -61,6 +63,95 @@ Samples getSamples(Cluster cluster,
                      long timeout)
       throws SamplingException;
 
+  /**
+   * Get the metric sample of the given topic partition and replica from the Kafka cluster.
+   *
+   * The samples include PartitionMetricSamples and BrokerMetricSamples.
+   *
+   * Due to the lack of direct metrics at partition level, Kafka Cruise Control needs to estimate the CPU
+   * utilization for each partition by using the following formula:
+   *
+   *  BROKER_CPU_UTIL = a * ALL_TOPIC_BYTES_IN_RATE + b * ALL_TOPIC_BYTES_OUT_RATE + c * ALL_FOLLOWER_BYTES_IN_RATE
+   *
+   *  LEADER_PARTITION_CPU_UTIL = a * LEADER_PARTITION_BYTES_IN + b * LEADER_PARTITION_BYTES_OUT
+   *
+   *  FOLLOWER_PARTITION_CPU_UTIL = c * LEADER_PARTITION_BYTES_IN
+   *
+   * Kafka Cruise Control needs to know the parameters of a, b and c for cost evaluation of leader and
+   * partition movement.
+   *
+   * @param metricSamplerOptions This class encapsulates all the arguments needed by MetricSampler to get samples.
+   * @return The PartitionMetricSample of the topic partition and replica id
+   */
+  Samples getSamples(MetricSamplerOptions metricSamplerOptions) throws SamplingException;
+
+  /**
+   * This class encapsulates the options that need to be passed to the {@link #getSamples(MetricSamplerOptions)}
+   * method. Future use-cases may choose to extend this class and add more options.
+   */
+  public class MetricSamplerOptions {
+    /**
+     * @param cluster the metadata of the cluster.
+     * @param assignedPartitions the topic partition
+     * @param startTimeMs the start time of the sampling period.
+     * @param endTimeMs the end time of the sampling period.
+     * @param mode The sampling mode.
+     * @param metricDef the metric definitions.
+     * @param timeout The sampling timeout to stop sampling even if there is more data to get.
+     */
+    public MetricSamplerOptions(Cluster cluster,
+                                Set<TopicPartition> assignedPartitions,
+                                long startTimeMs,
+                                long endTimeMs,
+                                SamplingMode mode,
+                                MetricDef metricDef,
+                                long timeout) {
+      this._cluster = cluster;
+      this._assignedPartitions = assignedPartitions;
+      this._startTimeMs = startTimeMs;
+      this._endTimeMs = endTimeMs;
+      this._mode = mode;
+      this._metricDef = metricDef;
+      this._timeout = timeout;
```

Comment:
Nit: Cruise Control codebase typically uses class fields without `this.` (not enforced) and with `_` prefix (enforced). For consistency with the typical use, can we drop the `this` prefix?